apicid/apicid.go

```diff
@@ -0,0 +1,61 @@
+package apicid
```

Comment:
It may sound weird but that is the exact idea I want to capture.  This package present a form of the Cid to use in the API struct.  The only reason it is here is because this type might need to be used outside of `go-ipfs`.